src/crypto/crypto_mbedtls.c

```diff
@@ -1,911 +1,4042 @@
 /*
- * SPDX-FileCopyrightText: 2020-2021 Espressif Systems (Shanghai) CO LTD
- * SPDX-FileCopyrightText: 2022 Nordic Semiconductor ASA
+ * crypto wrapper functions for mbed TLS
  *
- * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: 2022 Glenn Strauss <gstrauss@gluelogic.com>
+ * SPDX-License-Identifier: BSD-3-Clause
```

Comment:
You just cannot change the license and remove the old copyright statements, this is not how open source works.
cc: @dleach02

Reply:
it is total re-write. could you pls specify more how I shall do this, if current PR is incorrect?

Reply:
AFAIK, `Apache-2.0` doesn't allow to relicense the original code, the new code can be under new licesne but don't quote me on that, need some legal help here.

Do you want your changes to be `BSD-3-clause` only? If yes, then another way would be to a move to `port/mbedtls/crypto_mbedtls.c` and add a Kconfig to choose different implementations?

BTW, what about remaining files `crypto_mbedtls-ec.c`, `crypto_mbedtls-bignum.c` ...? Does your implementation don't need them?

Reply:
> I can add new BSD license under origin license. Is that looks okay to you?

Not sure if that would be allowed or not, others can chime in.

> crypto_mbedtls-ec.c, crypto_mbedtls-bignum.c I don't need them.

But WPA supplicant calls `crypto_bignum_*` APIs esp. in `sae.c` and `sae_pk.c` and also `crypto_ec_*` from `dpp.c` etc. How does this work?

Reply:
they are in crypto_mbedtls.c

Reply:
we are using like crypto_linux.c, maintaining them in one file.
and about the license, this PR is misleading. Because I removed it and replaced by a new file, just with the same name, but no same content.
Shall I raise a commit to delete it first and then create a new one, in split commits?
Or shall I name the new one crypto_mbedtls2.c, and keep origin file?

Reply:
Ah ok, missed that. So, I would definitely recommend submitting this as an alternative implementation:

```
backend_a/CURRENT MbedTLS
backend_b/This MBEDTLS

Kconfig: 

choice MBEDTLS_BACKEND
   
config a
config b
```

Not quite sure about naming of `a/b`, this way you are free to use your own license? @jukkar Is there a policy in Zephyr that `Apache-2.0` is the preferred or mandated License?

Reply:
> Ah ok, missed that. So, I would definitely recommend submitting this as an alternative implementation:
> 
> ```
> backend_a/CURRENT MbedTLS
> backend_b/This MBEDTLS
> 
> Kconfig: 
> 
> choice MBEDTLS_BACKEND
>    
> config a
> config b
> ```
> 

I agree with this. Is there a recommend name?

Reply:
Well, that is a tough decision, I don't want to use company names like `esp` and `nxp`, but then I don't know what to call them. May be `apache` or `bsd3`? but even that is undesirable. Let me think, meanwhile let's see if someone has any bright ideas here :).

Reply:
crypto_mbedtls2.c
crypto_mbedtls_alt.c

does either one look okay?

Reply:
Rather than renaming files, let's to folders, easier to fix in `cmake`.

We can start with `default` and `alt` if no one has any other suggestions.

Reply:
you mean create two folders: mbedtls_default and mbedtls_alt. and naming both file crypto_mbedtls.c?
I think I can add file and kconfig like

zephyr_library_sources_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_DEFAULT
    ${HOSTAP_SRC_BASE}/crypto/crypto_mbedtls.c
    ${HOSTAP_SRC_BASE}/crypto/tls_mbedtls.c
)

zephyr_library_sources_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT
    ${HOSTAP_SRC_BASE}/crypto/crypto_mbedtls_alt.c
    ${HOSTAP_SRC_BASE}/crypto/tls_mbedtls_alt.c
)

This looks easier and okay?

Reply:
Yes and for `CRYPTO_DEFAULT` we cann bignum, ec etc files.